core/state_transition.go

```diff
@@ -320,6 +320,10 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	// Set up the initial access list.
 	if rules.IsBerlin {
 		st.state.PrepareAccessList(msg.From(), msg.To(), vm.ActivePrecompiles(rules), msg.AccessList())
+		// EIP-3651 warm COINBASE
+		if rules.IsShanghai {
+			st.state.AddAddressToAccessList(st.evm.Context.Coinbase)
```

Comment:
This is not super-elegant, adding it like this means it's a journalled change. So it would be nicer to somehow integrate more closely in the "constructor" `PrepareAccessList`

Reply:
Also, there are a few other places where `PrepareAccessList` is invoked (e.g. `core/vm/runtime/runtime.go:Create`) which also needs to do this extra step, and are currently not. 
If we turn it into only one step, then we don't have to worry about missing the warm-coinbase extra-step

Reply:
How about changing the `PrepareAccessList` into
```
func (s *StateDB) PrepareAccessList(sender, coinbase common.Address, dst *common.Address, precompiles []common.Address, list types.AccessList, rules params.Rules) {

```

I.e: adding `coinbase common.Address`, and also adding `rules params.Rules`, so the method can add the coinbase if we're in shanghai.

Reply:
With the transient eip pr, the signature is changed (https://github.com/ethereum/go-ethereum/pull/26003/files#diff-c3757dc9e9d868f63bc84a0cc67159c1d5c22cc5d8c9468757098f0492e0658cR1071) , into 
```
func (s *StateDB) Prepare(rules params.Rules, sender common.Address, dst *common.Address, precompiles []common.Address, list types.AccessList) {
```
So you only need to add the coinbase